### ISSUE TITLE
Failed to subscribe to mastodon users except mastodon.xyz resource

### ISSUE THREAD

**Issue by mnd** [nextcloud/social#1118]
**Describe the bug**
After fresh Social 0.3.1 installation through stable nextcloud 19.0.4 I can successfully subscribe only to the mastodon.xyz users

**To Reproduce**
`occ social:account:follow mnd nextcloud@mastodon.xyz` and `occ social:account:follow -vvv --unfollow mnd Slaynash@mastodon.xyz` silently works
For other resources I got an error:
```
$ occ social:account:follow -vvv mnd russian_mastodon@mastodon.social

In TRequest.php line 292:
                                                                                                                                                                             
  [daita\MySmallPhpTools\Exceptions\RequestNetworkException (28)]                                                                                                            
  28 - Operation timed out after 10001 milliseconds with 352 out of -1 bytes received - {"protocols":["https","http"],"used_protocol":"https","host":"mastodon.social","url  
  ":"/.well-known/webfinger","timeout":10,"type":0,"data":{"resource":"acct:russian_mastodon@mastodon.social"}}                                                              
                                                                                                                                                                             

Exception trace:
  at /opt/nextcloud/apps/social/vendor/daita/my-small-php-tools/lib/Traits/TRequest.php:292
 OCA\Social\Service\CurlService->parseRequestResultCurl() at /opt/nextcloud/apps/social/vendor/daita/my-small-php-tools/lib/Traits/TRequest.php:270
 OCA\Social\Service\CurlService->parseRequestResult() at /opt/nextcloud/apps/social/vendor/daita/my-small-php-tools/lib/Traits/TRequest.php:127
 OCA\Social\Service\CurlService->doRequestOrig() at /opt/nextcloud/apps/social/lib/Service/CurlService.php:293
 OCA\Social\Service\CurlService->doRequest() at /opt/nextcloud/apps/social/vendor/daita/my-small-php-tools/lib/Traits/TRequest.php:76
 OCA\Social\Service\CurlService->retrieveJsonOrig() at /opt/nextcloud/apps/social/lib/Service/CurlService.php:265
 OCA\Social\Service\CurlService->retrieveJson() at /opt/nextcloud/apps/social/lib/Service/CurlService.php:138
 OCA\Social\Service\CurlService->webfingerAccount() at /opt/nextcloud/apps/social/lib/Service/CurlService.php:195
 OCA\Social\Service\CurlService->retrieveAccount() at /opt/nextcloud/apps/social/lib/Service/CacheActorService.php:218
 OCA\Social\Service\CacheActorService->getFromAccount() at /opt/nextcloud/apps/social/lib/Service/FollowService.php:137
 OCA\Social\Service\FollowService->followAccount() at /opt/nextcloud/apps/social/lib/Command/AccountFollowing.php:122
 OCA\Social\Command\AccountFollowing->execute() at /opt/nextcloud/3rdparty/symfony/console/Command/Command.php:255
 Symfony\Component\Console\Command\Command->run() at /opt/nextcloud/core/Command/Base.php:169
 OC\Core\Command\Base->run() at /opt/nextcloud/3rdparty/symfony/console/Application.php:1012
 Symfony\Component\Console\Application->doRunCommand() at /opt/nextcloud/3rdparty/symfony/console/Application.php:272
 Symfony\Component\Console\Application->doRun() at /opt/nextcloud/3rdparty/symfony/console/Application.php:148
 Symfony\Component\Console\Application->run() at /opt/nextcloud/lib/private/Console/Application.php:215
 OC\Console\Application->run() at /opt/nextcloud/console.php:100
 require_once() at /opt/nextcloud/occ:11

$ occ social:account:follow -vvv mnd cwebber@octodon.social

In TRequest.php line 292:
                                                                                                                                                                             
  [daita\MySmallPhpTools\Exceptions\RequestNetworkException (28)]                                                                                                            
  28 - Operation timed out after 10001 milliseconds with 537 out of -1 bytes received - {"protocols":["https"],"used_protocol":"https","host":"octodon.social","url":"/user  
  s/cwebber","timeout":10,"type":0,"data":[]}

***Same trace***
```

**Expected behavior**
Successful subscription

**Social app version:** 0.3.1

**Operating system**: CentOS Linux release 7.8.2003

**Web server:** nginx 1.16.1

**Database:** mariadb 5.5.65

**PHP version:** 7.3.11

**Nextcloud version:** 19.0.4

</details>

**Comment by mnd** [comment]
After additional investigation I observed that issue in the daita/my-small-php-tools code in curl requests implementation. Having small buffer in the https://github.com/daita/my-small-php-tools/blob/master/lib/Traits/TRequest.php#L155 and cURL library 7.29.0 from CentOS 7.0 I got same issue as was described https://core.trac.wordpress.org/ticket/46559 

Currently I increased BUFFERSIZE to 10K directly in the nextcloud/apps/social/vendor/daita/my-small-php-tools/lib/Traits/TRequest.php file.

Can you suggest correct solution for this issue?

**Comment by mnd** [comment]
Looks like it was solved by 68660884319e2bb03ebcbe7cdd26434b3473c50b
